flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/SnapshotUtils.java

```diff
@@ -259,6 +257,31 @@ protected static Optional<SnapshotTriggerType> shouldTriggerSnapshot(
         return Optional.empty();
     }
 
+    private static boolean shouldTriggerAutomaticSnapshot(
+            SnapshotType snapshotType, String automaticTriggerExpression, Instant lastTrigger) {
+        if (StringUtils.isBlank(automaticTriggerExpression)) {
+            return false;
+        }
+
+        boolean shouldTrigger =
+                shouldTriggerIntervalBasedSnapshot(
+                                snapshotType, automaticTriggerExpression, lastTrigger)
+                        || shouldTriggerCronBasedSnapshot(
```

Comment:
This condition means that theoretically still both can trigger. We should make sure that the condition guarantees mutual exclusion. So first check interval, then check cron based execution.

Reply:
I actually just realized that the logic is also false. We cannot rely on the true/false at all. It might be that both triggers are simply not due. We'll need to work with exceptions directly. I'll give it another go.

Reply:
Fixed the logic:
https://github.com/apache/flink-kubernetes-operator/pull/656/files#diff-deedaba89a69eefb0acf4c0fcbb250b5ec9415cd78a3323827a0ce0c162b8b49R261